tests/src/test/java/org/eclipse/hono/tests/registry/DeviceRegistrationAmqpIT.java

```diff
@@ -144,79 +140,22 @@ public void testAssertRegistrationFailsForDisabledDevice(final TestContext ctx)
-
-        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
-        final String gatewayId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
-        helper.registry.registerDevice(Constants.DEFAULT_TENANT, gatewayId)
-            .compose(ok -> helper.registry.registerDevice(Constants.DEFAULT_TENANT, deviceId, new JsonObject()
-                    .put(PROPERTY_VIA, gatewayId)))
-            .compose(ok -> deviceRegistryclient.getOrCreateRegistrationClient(Constants.DEFAULT_TENANT))
-            .compose(client -> client.assertRegistration(deviceId, gatewayId))
-            .setHandler(ctx.asyncAssertSuccess());
-    }
-
-    /**
-     * Verifies that the registry fails a disabled gateway's request to assert a
-     * device's registration status with a 403 error code.
-     * 
-     * @param ctx The vert.x test context.
-     */
-    @Test
-    public void testAssertRegistrationFailsForDisabledGateway(final TestContext ctx) {
-
-        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
-        final String gatewayId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
-        helper.registry.registerDevice(Constants.DEFAULT_TENANT, gatewayId, new JsonObject()
-                .put(RegistrationConstants.FIELD_ENABLED, false))
-            .compose(ok -> helper.registry.registerDevice(Constants.DEFAULT_TENANT, deviceId, new JsonObject()
-                    .put(PROPERTY_VIA, gatewayId)))
-            .compose(ok -> deviceRegistryclient.getOrCreateRegistrationClient(Constants.DEFAULT_TENANT))
-            .compose(client -> client.assertRegistration(deviceId, gatewayId))
-            .setHandler(ctx.asyncAssertFailure(t -> assertErrorCode(t, HttpURLConnection.HTTP_FORBIDDEN, ctx)));
-    }
-
-    /**
-     * Verifies that the registry fails a gateway's request to assert a
-     * device's registration status for which it is not authorized with a 403 error code.
-     * 
-     * @param ctx The vert.x test context.
-     */
-    @Test
-    public void testAssertRegistrationFailsForUnauthorizedGateway(final TestContext ctx) {
-
-        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
-        final String gatewayId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
-        helper.registry.registerDevice(Constants.DEFAULT_TENANT, gatewayId)
-            .compose(ok -> helper.registry.registerDevice(Constants.DEFAULT_TENANT, deviceId, new JsonObject()
-                    .put(PROPERTY_VIA, "not-" + gatewayId)))
-            .compose(ok -> deviceRegistryclient.getOrCreateRegistrationClient(Constants.DEFAULT_TENANT))
-            .compose(client -> client.assertRegistration(deviceId, gatewayId))
-            .setHandler(ctx.asyncAssertFailure(t -> assertErrorCode(t, HttpURLConnection.HTTP_FORBIDDEN, ctx)));
+                .compose(ok -> deviceRegistryclient.getOrCreateRegistrationClient(Constants.DEFAULT_TENANT))
```

Comment:
I guess you won't want to register a device here using the helper because the registry under test will probably not expose a corresponding HTTP based API, right?

Instead, the test should probably use the *pre-configured* device 4711 (which, BTW, should be defined as a constant IMHO).

Reply:
I think I noticed and removed this as you typed this comment :)

Reply:
@sophokles73 As i updated the JSON file to use pre-configured devices I added a few more devices to cover the various test cases. 
Do you suggest I define constants for all the devices ?

Reply:
> Do you suggest I define constants for all the devices ?

I think that would be a good idea. Maybe you can name the constants according to the characteristics of the device(s), e.g. *DEVICE_ID_ENABLED* and *DEVICE_ID_DISABLED*?

I think that the mandatory tests should also include a test for asserting a non-existing device ID.

Reply:
> I think that the mandatory tests should also include a test for asserting a non-existing device ID.
There is `testAssertRegistrationFailsForUnknownDevice` which verifies that the service reply 404 for a non-existing device.

Reply:
oh, sure. My bad ...

Reply:
>Maybe you can name the constants according to the characteristics of the device(s), e.g. _DEVICE_ID_ENABLED_ and _DEVICE_ID_DISABLED_?

I updated the PR with that.